src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Transport/IPluginDownloader.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.IO;
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Transport
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public interface IPluginDownloader
```

Comment:
Not every source has a notion of "download." E.g. installing from a file on the file system.

I think this interface would make sense to remove, and instead have a method on `IAvailablePlugin` like 

```cs
public interface IAvailablePlugin
{
    void InstallTo(string folderPath);
}
```

The idea being that a `PluginDiscoverer` returns custom `IAvailablePlugin` implementations that know how to install itself. The manager tells it "install to this location on disk"

Reply:
The downloader is removed as an interface. The return type remained to be a stream to a PluginPackage to be handled by the installer. The usage of it will be shown in the next PR.